pkg/reconciler/revision/cruds.go

```diff
@@ -71,8 +71,13 @@ func (c *Reconciler) checkAndUpdateDeployment(ctx context.Context, rev *v1.Revis
 	desiredDeployment := have.DeepCopy()
 	desiredDeployment.Spec = deployment.Spec
 
-	// Carry over new labels.
+	// Carry over new labels and annotations at deployment level.
 	desiredDeployment.Labels = kmeta.UnionMaps(deployment.Labels, desiredDeployment.Labels)
+	desiredDeployment.Annotations = kmeta.UnionMaps(deployment.Annotations, desiredDeployment.Annotations)
+
+	// Carry over template-level labels and annotations (e.g., kubectl.kubernetes.io/restartedAt from rollout restart).
+	desiredDeployment.Spec.Template.Labels = kmeta.UnionMaps(deployment.Spec.Template.Labels, have.Spec.Template.Labels)
+	desiredDeployment.Spec.Template.Annotations = kmeta.UnionMaps(deployment.Spec.Template.Annotations, have.Spec.Template.Annotations)
```

Comment:
One thing I learned from working on - https://github.com/knative/serving/pull/16186/files

If we union the `have` map with the `desired` one then we end up losing the ability to delete things.

Thus we should filter out labels/annotations we control from the `have` map.

I think we can just remove `*.knative.dev` and we also have an `app` label (though I don't expect us to remove that)

Everything else should be preserved. See my other PR as a starting point - there I assume we only care about specific autoscaling labels for the pod autoscaler.